src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -64,55 +62,26 @@
  * about these suffixes are added.
  */
 public class SkroutzGreekStemmer {
-  public final static String DEFAULT_STOPWORD_FILE = "stopwords.txt";
-  private final CharArraySet stopwords;
-  private Map<String, char[]> stepZeroExceptions;
+  private CharArraySet protectedWords = CharArraySet.EMPTY_SET;
+  private CharArrayMap<char[]> stepZeroExceptions = CharArrayMap
+          .emptyMap();
 
-  public SkroutzGreekStemmer(final CharArraySet stopwords) {
-    this.stopwords = stopwords;
+  public SkroutzGreekStemmer() {
     try {
+      this.protectedWords = loadProtected("protected_words");
       this.stepZeroExceptions = loadExceptions("step_0_exceptions");
     } catch(Exception ex) {
       System.err.println("Failed to load exceptions from yaml file");
-      ex.printStackTrace();
-      this.stepZeroExceptions =  new HashMap(0);
-    }
-  }
-
-  public SkroutzGreekStemmer() {
-    this(SkroutzGreekStemmer.getDefaultStopSet());
-  }
-
-  public static final CharArraySet getDefaultStopSet() {
-    return DefaultSetHolder.DEFAULT_SET;
-  }
-
-  private static class DefaultSetHolder {
-    private static final CharArraySet DEFAULT_SET;
-
-    static {
-      try {
-        DEFAULT_SET = loadStopwordSet(
-            SkroutzGreekStemmer.class.getResourceAsStream(DEFAULT_STOPWORD_FILE),
-            Lucene.VERSION);
-      } catch (IOException ex) {
-        // default set should always be present as it is part of the
-        // distribution (JAR)
-        throw new RuntimeException("Unable to load default stopword set");
-      }
     }
   }
 
   public int stem(char s[], int len) {
-    // Too short or a stopword
-    if (len < 3 || stopwords.contains(s, 0, len))
+    // Too short or a protected word
+    if (len < 3 || protectedWords.contains(s, 0, len))
```

Comment:
In general I prefer explicitly adding parentheses around all boolean expressions: ((len <3) || (protectedWords.contains(s, 0, len))